src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,357 @@
+        with catmaid:// or catmaids:// for http and https protocols respectively.
+    return (AcquisitionData): an opened Catmaid instance
+    """
+    if any(url.startswith(p) for p in PREFIXES):
+        url = re.sub(r"catmaid([s]?)://", r"http\1://", url, 1)
+    else:
+        raise IOError("URL should start with catmaid:// or catmaids://")
+    urlo = urlparse.urlparse(url)
+    parameters = urlparse.parse_qs(urlo.query)
+    base_url = '{uri.scheme}://{uri.netloc}{uri.path}'.format(uri=urlo).rstrip("/")
+    project_id = int(parameters["pid"][0]) if "pid" in parameters.keys() else None
+    stack_id = int(parameters["sid0"][0]) if "sid0" in parameters.keys() else None
+
+    if project_id is None or stack_id is None:
+        try:
+            token, _, _ = read_config_file(base_url, token=True)
+            auth = CatmaidApiTokenAuth(token) if token else None
+            project_url = "{url}/projects/".format(url=base_url)
+            response = requests.get(project_url, auth=auth)
+            if response.status_code == 401:
+                raise AuthenticationError("Wrong token while getting project info at {}".format(project_url))
+            else:
+                response.raise_for_status()
+            project_info = response.json()
+            if project_info == [] and token is None:
+                raise AuthenticationError(
+                    "No project at {}, this Catmaid instance does not contain projects"
+                    "or a token should have been provided.".format(project_url))
+            project_id = project_info[0]["id"] if project_id is None else project_id
+            # loop through the projects to get the info of the project matching the project id.
+            project = [p for p in project_info if p["id"] == project_id]
+            stack_id = project[0]["stacks"][0]["id"] if stack_id is None else stack_id
+        except Exception:
+            # Try with project id and stack id is 1 if the project info cannot be accessed.
+            project_id = 1 if project_id is None else project_id
+            stack_id = 1 if stack_id is None else stack_id
+        logging.info(
+            "Project id and/or stack id not entered, using project {} and stack {}.".format(project_id, stack_id))
+    return AcquisitionDataCatmaid(base_url, project_id, stack_id)
+
+
+class DataArrayShadowPyramidalCatmaid(DataArrayShadow):
+    """
+    This class implements the read of a Pyramidal Catmaid instance.
+    """
+
+    def __init__(self, stack_info, base_url):
+        """
+        Constructor
+        stack_info (dict): information about the Catmaid stack and tiles in the stack.
+        base_url (str): URL where the Catmaid instance is hosted.
+        dtype (numpy.dtype): the data type
+        metadata (dict str->val): The metadata
+        """
+        shape = (stack_info["dimension"]["x"], stack_info["dimension"]["y"])
```

Comment:
Just a comment: let's be aware that such code is a little too optimistic, as if the server ever returns a json file which is not following its expectations, it'll immediately entirely fail with some KeyError exceptions which will mean nothing to the user...
That said, I'm fine with this for now, as this code is a first iteration, and the goal is to use it only on a server that we control. If we ever want to use it on a more "random" server we probably should be more defensive.